modules/navier_stokes/src/executioners/SIMPLENonlinearAssembly.C

```diff
@@ -478,6 +478,9 @@ SIMPLENonlinearAssembly::execute()
       // Resdiual index
       size_t residual_index = 0;
 
+      // Execute all objects tagged as nonlinear
+      _problem.execute(EXEC_NONLINEAR);
```

Comment:
Please add a few comments for this. It is rather unconventional, but important for things to work!